lib/td/client/api/job.rb

```diff
@@ -287,6 +248,114 @@ def query(q, type=:hive, db=nil, result_url=nil, priority=nil, retry_limit=nil,
 
   private
 
+  def validate_content_length(response, current_total_chunk_size)
+    if expected_size = response.header['Content-Range'][0]
+      expected_size = expected_size[/\d+$/].to_i
+    elsif expected_size = response.header['Content-Length'][0]
+      expected_size = expected_size.to_i
+    end
+
+    if expected_size.nil?
+    elsif current_total_chunk_size < expected_size
+      # too small
+      raise EOFError, 'httpclient IncompleteError'
+    elsif current_total_chunk_size < expected_size
+      # too large
+      raise_error("Get job result failed", res)
+    end
+  end
+
+  def job_result_download(job_id, format='msgpack', decode_p=true)
```

Comment:
`decode_p` is a bit of obscure name. It is called `autodecode` [in a spec](https://github.com/treasure-data/td-client-ruby/pull/85/files#diff-5bd124e8c2691a58439151c8acde3692R783).